### PR TITLE
Update: Add `unnecessary` option to `quote-props`

### LINKED ISSUE #3381: Lost use case with `no-reserved-keys` to `quote-props` merge
Previously I had `quote-props` turned off and wanted errors for `no-reserved-keys` but that now seems impossible. In other words, I don't care about property quoting, except if it's a keyword.

Another way to look at it is wanting `as-needed` to have an option to ignore unnecessarily quoted properties.

### PR DESCRIPTION
**Pull request by BYK**
Fixes #3381

### CODE REVIEW

**Comment by gyandeeps** [comment]
Since we are adding a new option, we need to update the docs also.

**Comment by BYK** [comment]
@gyandeeps - Yeah I keep forgetting about docs :( Thanks for the reminder, updated.

That said I'm wondering if this configuration is the easiest. May be we should make `keyword: "as-needed"` instead of adding a new property. //cc @nzakas

**Comment by nzakas** [comment]
This doesn't have to do with keywords. The concern is about turning off warnings for non-keywords, so I think the option name is fine.

File: tests/lib/rules/quote-props.js [hunk 1]
```diff
@@ -132,5 +132,11 @@ ruleTester.run("quote-props", rule, {
         errors: [{
             message: "Unquoted reserved word `while` used as key.", type: "Property"
         }]
+    }, {
+        code: "({'unnecessary': 1, if: 0})",
```
Comment (nzakas) [inline tests/lib/rules/quote-props.js]: Can you add a valid test as well?

Comment (BYK) [inline tests/lib/rules/quote-props.js]: Done.

**Comment by nzakas** [comment]
LGTM

**Pull request merged**